test/Aux.hs

```diff
@@ -0,0 +1,123 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Aux (runBuiltin) where
+
+import Control.Monad.Error.Class (liftEither)
+import Data.Aeson qualified as Aeson
+import Data.Bifunctor (Bifunctor (first))
+import Data.Coerce (Coercible, coerce)
+import Data.Default.Class (Default (def))
+import Data.Functor (void)
+import Data.Kind (Type)
+import Data.Text (Text)
+import PlutusCore
+  ( DefaultFun,
+    DefaultUni,
+    Error (FreeVariableErrorE),
+    Name,
+    Term,
+    TyName,
+    TypeCheckConfig (TypeCheckConfig),
+    builtinMeaningsToTypes,
+    deBruijnTerm,
+    defKindCheckConfig,
+    inferType,
+    runQuoteT,
+  )
+import PlutusCore.Compiler.Erase (eraseTerm)
+import PlutusCore.Evaluation.Machine.ExBudget
+  ( ExBudget,
+    ExRestrictingBudget
+      ( ExRestrictingBudget
+      ),
+  )
+import PlutusCore.Evaluation.Machine.ExBudgetingDefaults qualified as UPLC
+import PlutusCore.MkPlc (TermLike (builtin), mkIterAppNoAnn)
+import PlutusCore.Pretty (prettyPlcClassicDebug)
+import System.Exit (die)
+import System.FilePath ((</>))
+import UntypedPlutusCore qualified as UPlc
+import UntypedPlutusCore.Evaluation.Machine.Cek qualified as UPlc
+
+runBuiltin ::
+  String -> -- Prefix
+  FilePath -> -- Path of a directory, where all the artifacts will be written to
+  Maybe ExBudget ->
+  DefaultFun -> -- A builtin function
+  [Term TyName Name DefaultUni DefaultFun ()] -> -- Arguments to apply to the builtin function
+  IO ()
+runBuiltin name outputDir budget builtinFn args = do
+  let term = appBulitinFun builtinFn args
+
+  (checkedTerm, (evalResult, evalBudget, _evalLog)) <-
+    either
+      (die . (\err -> "(" <> name <> ") " <> "Failed to type check: " <> err) . show)
```

Comment:
I think we can be a bit more principled here. We can easily work in `ExceptT` on top of `IO`.

Reply:
I don't see any harm in doing so, although now that you mention it, it probably doesn't matter very much.

Reply:
Implemented in https://github.com/mlabs-haskell/plutus-conformance/pull/8/commits/f0b8c56844e5af3e69d3729d8d074075381f6fff, but using `MonadError` instead of `ExceptT`.